scene/main/viewport.cpp

```diff
@@ -2904,6 +2904,7 @@ bool Viewport::_sub_windows_forward_input(const Ref<InputEvent> &p_event) {
 
 void Viewport::push_input(const Ref<InputEvent> &p_event, bool p_local_coords) {
 	ERR_FAIL_COND(!is_inside_tree());
+	ERR_FAIL_NULL_MSG(p_event, "p_event cannot be null.");
```

Comment:
That error message isn't much more explicit than the default error message for `ERR_FAIL_NULL`, so I would just do `ERR_FAIL_NULL(p_event);`.

Reply:
Shouldn't it be either `ERR_FAIL_COND(p_event.is_null());` or `ERR_FAIL_NULL(p_event.ptr());` instead? `p_event` is a reference.

Reply:
Indeed, good catch. `ERR_FAIL_COND(p_event.is_null());` would be the common way we check this.